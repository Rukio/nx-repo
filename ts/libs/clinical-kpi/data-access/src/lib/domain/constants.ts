export const LEADS_PROVIDERS_API_PATH = 'providers';
