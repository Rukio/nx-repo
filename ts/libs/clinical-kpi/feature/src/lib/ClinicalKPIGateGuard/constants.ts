export const CLINICAL_KPI_FEATURE_GATE_NAME = 'kpi_dashboard_integration';
export const CLINICAL_KPI_FEATURE_GATE_ERROR_TEXT =
  'This feature has not been released in your market yet!';
