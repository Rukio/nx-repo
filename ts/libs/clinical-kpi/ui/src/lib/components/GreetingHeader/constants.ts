export const MORNING_GREETING = 'Good Morning';
export const AFTERNOON_GREETING = 'Good Afternoon';
export const EVENING_GREETING = 'Good Evening';
export const STATION_URL_QA = 'https://qa.*company-data-covered*.com/';
