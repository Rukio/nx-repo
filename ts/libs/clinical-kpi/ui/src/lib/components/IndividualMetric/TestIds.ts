export const INDIVIDUAL_METRIC_TEST_IDS = {
  TYPE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-type`,
  VALUE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-value`,
  VALUE_CHANGE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-change`,
  PERFORMANCE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-performance`,
  GOAL: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-goal`,
};
