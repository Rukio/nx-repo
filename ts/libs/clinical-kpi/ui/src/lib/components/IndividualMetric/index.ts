export { default as IndividualMetric } from './IndividualMetric';
export { INDIVIDUAL_METRIC_TEST_IDS } from './TestIds';
