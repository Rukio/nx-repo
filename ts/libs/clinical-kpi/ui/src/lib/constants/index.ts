import { SxStylesValue } from '@*company-data-covered*/design-system';

export enum Metrics {
  OnSceneTime = 'OnSceneTime',
  ChartClosure = 'ChartClosure',
  SurveyCapture = 'SurveyCapture',
  NPS = 'NPS',
  OnTaskPercent = 'OnTaskPercent',
  Escalation = 'Escalation',
  ABX = 'ABX',
}

export enum UnitsOfMeasure {
  None = 'None',
  Percent = 'Percent',
  Minutes = 'Minutes',
}

export enum TrendDirection {
  NEGATIVE = 'NEGATIVE',
  NEUTRAL = 'NEUTRAL',
  POSITIVE = 'POSITIVE',
}

export const METRICS_UNIT_OF_MEASURE: Record<Metrics, UnitsOfMeasure> = {
  ChartClosure: UnitsOfMeasure.Percent,
  SurveyCapture: UnitsOfMeasure.Percent,
  OnSceneTime: UnitsOfMeasure.Minutes,
  NPS: UnitsOfMeasure.None,
  OnTaskPercent: UnitsOfMeasure.Percent,
  Escalation: UnitsOfMeasure.Percent,
  ABX: UnitsOfMeasure.Percent,
};

export const TREND_DIRECTION_STYLES: Record<TrendDirection, SxStylesValue> = {
  POSITIVE: {
    color: 'success.main',
  },
  NEUTRAL: {
    color: 'text.disabled',
  },
  NEGATIVE: {
    color: 'error.main',
  },
};

export enum SortDirection {
  Ascending = 'Ascending',
  Descending = 'Descending',
}

export enum ValueFontColor {
  ValueChangeGreenCell = 'valueChangeGreenCell',
  ValueChangeRedCell = 'valueChangeRedCell',
  ValueChangeGrayCell = 'valueChangeGrayCell',
}

export const METRICS_DISPLAY_NAME: Record<Metrics, string> = {
  OnSceneTime: 'On Scene Time',
  ChartClosure: 'Chart Closure',
  SurveyCapture: 'Survey Capture',
  NPS: 'NPS',
  OnTaskPercent: 'On Task %',
  Escalation: 'Escalation Rate',
  ABX: 'ABX Prescribing Rate',
};

export const METRICS_DEFAULT_SORT_DIRECTION: Record<Metrics, SortDirection> = {
  OnSceneTime: SortDirection.Ascending,
  ChartClosure: SortDirection.Descending,
  SurveyCapture: SortDirection.Descending,
  NPS: SortDirection.Descending,
  OnTaskPercent: SortDirection.Descending,
  Escalation: SortDirection.Descending,
  ABX: SortDirection.Descending,
};

export const LEADS_VIEW_INDIVIDUAL_VISIBILITY =
  'leads_view_individual_visibility';

export const DEFAULT_ERROR_ALERT_TEXT =
  'Content failed to load. Please refresh the page.';
