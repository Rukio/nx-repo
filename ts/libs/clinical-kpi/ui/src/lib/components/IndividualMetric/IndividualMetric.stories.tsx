import { StoryFn, Meta } from '@storybook/react';
import { IndividualMetric } from './';
import { Metrics } from '../../constants';

export default {
  component: IndividualMetric,
  title: 'IndividualMetric',
} as Meta<typeof IndividualMetric>;

const Template: StoryFn<typeof IndividualMetric> = (args) => (
  <IndividualMetric {...args} />
);

export const Primary = Template.bind({});

Primary.args = {
  type: Metrics.OnSceneTime,
  value: 33,
  valueChange: 12,
  goal: 21,
};
