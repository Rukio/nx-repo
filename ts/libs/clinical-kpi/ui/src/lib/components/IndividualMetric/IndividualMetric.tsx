import {
  Card,
  Box,
  makeSxStyles,
  Typography,
} from '@*company-data-covered*/design-system';
import { INDIVIDUAL_METRIC_TEST_IDS } from './TestIds';
import { FC, useMemo } from 'react';
import { Metrics, METRICS_DISPLAY_NAME } from '../../constants';
import {
  formatMetricValue,
  formatMetricValueChange,
} from '../../utils/metricsUtils';

export type IndividualMetricProps = {
  type: Metrics;
  value: number;
  valueChange: number;
  goal?: number;
  rank?: number | string;
  testIdPrefix: string;
  possessiveOfMetric?: string;
  totalProviders?: string;
};

const makeStyles = () =>
  makeSxStyles({
    container: {
      display: 'flex',
      alignItems: 'left',
      flexDirection: 'column',
      height: '100%',
      width: '100%',
      borderRadius: '8px',
      padding: 2,
    },
    BoxWrapper: {
      display: 'flex',
      flexDirection: 'row',
      p: 1,
    },
    textWrapper: (theme) => ({
      p: 1,
      color: theme.palette.grey[700],
    }),
    rankTextWrapper: (theme) => ({
      color: theme.palette.text.secondary,
      p: 1,
    }),
    value: {
      fontSize: '1.5rem',
    },
    valueChange: {
      fontFamily: 'Open Sans',
      my: 'auto',
      mx: 1,
    },
  });

export const getPerformanceLineText = (
  type: keyof typeof Metrics,
  numericValueChange: number,
  displayValueChange: string,
  possessiveOfMetric?: string
) => {
  const performanceLineStart = `Since last week, ${possessiveOfMetric} ${METRICS_DISPLAY_NAME[type]}`;

  if (numericValueChange === 0) {
    return `${performanceLineStart} did not change`;
  } else if (numericValueChange > 0) {
    return `${performanceLineStart} went up ${displayValueChange.replace(
      '+',
      ''
    )}`;
  }

  return `${performanceLineStart} went down ${displayValueChange.replace(
    '-',
    ''
  )}`;
};

const IndividualMetric: FC<IndividualMetricProps> = ({
  type,
  value,
  valueChange,
  goal,
  rank,
  testIdPrefix,
  possessiveOfMetric,
  totalProviders,
}) => {
  const styles = makeStyles();

  const { displayValue } = useMemo(
    () => formatMetricValue({ type, value }),
    [type, value]
  );
  const {
    displayValue: displayValueChange,
    numericValue: numericValueChange,
    styles: valueChangeStyles,
  } = useMemo(
    () => formatMetricValueChange({ type, value: valueChange }),
    [type, valueChange]
  );

  const performanceLineText = getPerformanceLineText(
    type,
    numericValueChange,
    displayValueChange,
    possessiveOfMetric
  );

  return (
    <Card sx={styles.container}>
      <Box sx={styles.BoxWrapper}>
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.TYPE(testIdPrefix)}
          variant="subtitle1"
        >
          {METRICS_DISPLAY_NAME[type]}
        </Typography>
      </Box>

      <Box sx={styles.BoxWrapper}>
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.VALUE(testIdPrefix)}
          variant="h4"
          sx={styles.value}
        >
          {displayValue}
        </Typography>
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.VALUE_CHANGE(testIdPrefix)}
          sx={[styles.valueChange, valueChangeStyles]}
          variant="label"
        >
          {displayValueChange}
        </Typography>
      </Box>
      {possessiveOfMetric && performanceLineText && (
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.PERFORMANCE(testIdPrefix)}
          sx={styles.textWrapper}
          variant="label"
        >
          {performanceLineText}
        </Typography>
      )}
      {goal && (
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.GOAL(testIdPrefix)}
          variant="body2"
          sx={styles.textWrapper}
        >
          Goal: {goal}
        </Typography>
      )}
      {rank && totalProviders && (
        <Typography
          data-testid={INDIVIDUAL_METRIC_TEST_IDS.GOAL(testIdPrefix)}
          variant="body2"
          sx={styles.rankTextWrapper}
        >
          Rank {rank} out of {totalProviders}
        </Typography>
      )}
    </Card>
  );
};

export default IndividualMetric;
