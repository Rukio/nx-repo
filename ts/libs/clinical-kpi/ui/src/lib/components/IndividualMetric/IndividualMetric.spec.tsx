import { render, screen } from '@testing-library/react';
import IndividualMetric, {
  getPerformanceLineText,
  IndividualMetricProps,
} from './IndividualMetric';
import { Metrics } from '../../constants';

const props: IndividualMetricProps = {
  type: Metrics.OnSceneTime,
  value: 57.7,
  valueChange: 14.27,
  goal: 21,
  testIdPrefix: 'text-prefix',
  possessiveOfMetric: 'your',
};

describe('IndividualMetric', () => {
  it('should render correctly for OnSceneTime type', () => {
    const { asFragment } = render(
      <IndividualMetric {...props} type={Metrics.OnSceneTime} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly for SurveyCapture type', () => {
    const { asFragment } = render(
      <IndividualMetric {...props} type={Metrics.SurveyCapture} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly for ChartClosure type', () => {
    const { asFragment } = render(
      <IndividualMetric {...props} type={Metrics.ChartClosure} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly for NPS type', () => {
    const { asFragment } = render(
      <IndividualMetric {...props} type={Metrics.NPS} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly for different possessiveOfMetric', () => {
    const { asFragment } = render(
      <IndividualMetric
        {...props}
        possessiveOfMetric="Market's"
        type={Metrics.NPS}
      />
    );
    const metricText = screen.getByTestId(
      'text-prefix-individual-metric-performance'
    );
    expect(metricText).toHaveTextContent(
      "Since last week, Market's NPS went up 14.27"
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('unit tests', () => {
  describe('getPerformanceLineText()', () => {
    test.each([
      {
        type: Metrics.OnSceneTime,
        numericValueChange: 10.12,
        displayValueChange: '10.12 mins',
        expectedValue: 'Since last week, your On Scene Time went up 10.12 mins',
      },
      {
        type: Metrics.ChartClosure,
        numericValueChange: 20.12,
        displayValueChange: '20.12%',
        expectedValue: 'Since last week, your Chart Closure went up 20.12%',
      },
      {
        type: Metrics.SurveyCapture,
        numericValueChange: -30.12,
        displayValueChange: '-30.12%',
        expectedValue: 'Since last week, your Survey Capture went down 30.12%',
      },
      {
        type: Metrics.NPS,
        numericValueChange: 40.12345,
        displayValueChange: '40.12',
        expectedValue: 'Since last week, your NPS went up 40.12',
      },
      {
        type: Metrics.NPS,
        numericValueChange: 0,
        displayValueChange: '0',
        expectedValue: 'Since last week, your NPS did not change',
      },
    ])(
      `getPerformanceLine(type: keyof typeof Metrics, valueChange: number, differenceLine: string)`,
      ({ type, numericValueChange, displayValueChange, expectedValue }) => {
        expect(
          getPerformanceLineText(
            type,
            numericValueChange,
            displayValueChange,
            'your'
          )
        ).toEqual(expectedValue);
      }
    );
  });
});
