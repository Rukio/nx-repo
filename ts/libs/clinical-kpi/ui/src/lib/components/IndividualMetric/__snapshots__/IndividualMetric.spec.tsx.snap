// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualMetric should render correctly for ChartClosure type 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-5 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #2e7d32;
}

.emotion-6 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
        data-testid="text-prefix-individual-metric-type"
      >
        Chart Closure
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-4"
        data-testid="text-prefix-individual-metric-value"
      >
        57.7%
      </h4>
      <span
        class="MuiTypography-root MuiTypography-label emotion-5"
        data-testid="text-prefix-individual-metric-change"
      >
        +14.27%
      </span>
    </div>
    <span
      class="MuiTypography-root MuiTypography-label emotion-6"
      data-testid="text-prefix-individual-metric-performance"
    >
      Since last week, your Chart Closure went up 14.27%
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-7"
      data-testid="text-prefix-individual-metric-goal"
    >
      Goal: 21
    </p>
  </div>
</DocumentFragment>
`;

exports[`IndividualMetric should render correctly for NPS type 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-5 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #2e7d32;
}

.emotion-6 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
        data-testid="text-prefix-individual-metric-type"
      >
        NPS
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-4"
        data-testid="text-prefix-individual-metric-value"
      >
        57.7
      </h4>
      <span
        class="MuiTypography-root MuiTypography-label emotion-5"
        data-testid="text-prefix-individual-metric-change"
      >
        +14.27
      </span>
    </div>
    <span
      class="MuiTypography-root MuiTypography-label emotion-6"
      data-testid="text-prefix-individual-metric-performance"
    >
      Since last week, your NPS went up 14.27
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-7"
      data-testid="text-prefix-individual-metric-goal"
    >
      Goal: 21
    </p>
  </div>
</DocumentFragment>
`;

exports[`IndividualMetric should render correctly for OnSceneTime type 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-5 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #d32f2f;
}

.emotion-6 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
        data-testid="text-prefix-individual-metric-type"
      >
        On Scene Time
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-4"
        data-testid="text-prefix-individual-metric-value"
      >
        57.7 mins
      </h4>
      <span
        class="MuiTypography-root MuiTypography-label emotion-5"
        data-testid="text-prefix-individual-metric-change"
      >
        +14.27 mins
      </span>
    </div>
    <span
      class="MuiTypography-root MuiTypography-label emotion-6"
      data-testid="text-prefix-individual-metric-performance"
    >
      Since last week, your On Scene Time went up 14.27 mins
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-7"
      data-testid="text-prefix-individual-metric-goal"
    >
      Goal: 21
    </p>
  </div>
</DocumentFragment>
`;

exports[`IndividualMetric should render correctly for SurveyCapture type 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-5 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #2e7d32;
}

.emotion-6 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
        data-testid="text-prefix-individual-metric-type"
      >
        Survey Capture
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-4"
        data-testid="text-prefix-individual-metric-value"
      >
        57.7%
      </h4>
      <span
        class="MuiTypography-root MuiTypography-label emotion-5"
        data-testid="text-prefix-individual-metric-change"
      >
        +14.27%
      </span>
    </div>
    <span
      class="MuiTypography-root MuiTypography-label emotion-6"
      data-testid="text-prefix-individual-metric-performance"
    >
      Since last week, your Survey Capture went up 14.27%
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-7"
      data-testid="text-prefix-individual-metric-goal"
    >
      Goal: 21
    </p>
  </div>
</DocumentFragment>
`;

exports[`IndividualMetric should render correctly for different possessiveOfMetric 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-5 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #2e7d32;
}

.emotion-6 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
        data-testid="text-prefix-individual-metric-type"
      >
        NPS
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-4"
        data-testid="text-prefix-individual-metric-value"
      >
        57.7
      </h4>
      <span
        class="MuiTypography-root MuiTypography-label emotion-5"
        data-testid="text-prefix-individual-metric-change"
      >
        +14.27
      </span>
    </div>
    <span
      class="MuiTypography-root MuiTypography-label emotion-6"
      data-testid="text-prefix-individual-metric-performance"
    >
      Since last week, Market's NPS went up 14.27
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-7"
      data-testid="text-prefix-individual-metric-goal"
    >
      Goal: 21
    </p>
  </div>
</DocumentFragment>
`;
